OsmAnd/src/net/osmand/plus/myplaces/tracks/filters/SmartFolderHelper.kt

```diff
@@ -251,8 +240,27 @@ class SmartFolderHelper(val app: OsmandApplication) {
 	}
 
 	fun getAllAvailableTrackItems(): HashSet<TrackItem> {
-		val items = HashSet<TrackItem>()
-		items.addAll(allAvailableTrackItems)
-		return items
+		return HashSet(allAvailableTrackItems)
+	}
+
+	private class SmartFoldersUpdateTask(
+		private val app: OsmandApplication,
+	) : AsyncTask<Void, Void, Void?>() {
+
+		@Deprecated("Deprecated in Java")
+		override fun doInBackground(vararg params: Void): Void? {
+			app.smartFolderHelper.run {
+				readSettings()
+				for (folder in smartFolderCollection) {
+					updateSmartFolderItems(folder)
+				}
+				notifyUpdateListeners()
```

Comment:
Probably better to move it to onPostExecute because there is no annotation about workerThread and view changes in non-UI thread lead to crashes